snaxc/ir/dart/scheduler.py

```diff
@@ -122,12 +123,38 @@ def is_pure_output_stationary(template: Template, schedule: Schedule):
     return first_reduction_idx > last_parallel_idx
 
 
+def is_memory_flexible_enough(
+    template: Template, schedule: Schedule, element_sizes: Sequence[int]
+):
+    """
+    Checks whether the TCDM flexibility is sufficient to actually execute
+    the schedule.
+
+    There must be one spatial stride of 1 that doesn't need more fine-grained
+    temporal access within one bank, such that that dimension can be packed together.
+    """
+    TCDM_BANK_WIDTH = 8
+    # this check only makes sense if there are temporal dimensions:
```

Comment:
I'm a bit confused by this phrasing, how about this?
```suggestion
    # If there are no temporal dimensions, this check is not necessary and the memory is for sure flexible enough:
```